Tests fail when samples are disabled

Building the tests without samples results in failure, as GLFW (among other things) will be missing:
```bash
Vulkan-Hpp/tests/UniqueHandle/../../samples/utils/utils.hpp:21:10: fatal error: 'GLFW/glfw3.h' file not found
   21 | #include <GLFW/glfw3.h>
```

Reproducible via:
```cmake
VULKAN_HPP_SAMPLES_BUILD OFF
VULKAN_HPP_TESTS_BUILD ON
VULKAN_HPP_TESTS_BUILD_ONLY_DYNAMIC ON
```

The tests for `UniqueHandle` require more targets to be available, which are (for samples at least) added in the CMake right here:

https://github.com/KhronosGroup/Vulkan-Hpp/blob/4266ef3040eae8101eef318756bea2549d942158/CMakeLists.txt#L526-L536

I'm not sure how you would want to handle this @asuessenbach so I didn't open a PR. We could simply check for whether _either_ samples or tests are being built, or simplify the `UniqueHandle` tests?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -529,13 +529,19 @@ function( vulkan_hpp__setup_test )
 	target_link_libraries( ${TARGET_NAME} PRIVATE ${TARGET_LIBRARIES} )
 endfunction()
 
-if( VULKAN_HPP_SAMPLES_BUILD )
+if( VULKAN_HPP_SAMPLES_BUILD OR VULKAN_HPP_TESTS_BUILD )
 	# external libraries
 	add_subdirectory( glm )
 	set( GLFW_BUILD_EXAMPLES OFF )
 	set( GLFW_BUILD_TESTS OFF )
 	add_subdirectory( glfw )
 	add_subdirectory( glslang )
+
+	add_subdirectory( samples/utils )
+	add_subdirectory( RAII_Samples/utils )
+endif()
+
+if( VULKAN_HPP_SAMPLES_BUILD )
 	# samples
 	add_subdirectory( samples )
 	add_subdirectory( RAII_Samples )

--- a/RAII_Samples/CMakeLists.txt
+++ b/RAII_Samples/CMakeLists.txt
@@ -12,7 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-add_subdirectory( utils )
 add_subdirectory( 01_InitInstance )
 add_subdirectory( 02_EnumerateDevices )
 add_subdirectory( 03_InitDevice )

--- a/samples/CMakeLists.txt
+++ b/samples/CMakeLists.txt
@@ -12,7 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-add_subdirectory( utils )
 add_subdirectory( 01_InitInstance )
 add_subdirectory( 02_EnumerateDevices )
 add_subdirectory( 03_InitDevice )

